app/controllers/plates_controller.rb

```diff
@@ -8,35 +8,82 @@ class PlatesController < LabwareController
   before_action :check_for_current_user!, only: %i[update fail_wells] # rubocop:todo Rails/LexicallyScopedActionFilter
 
   def fail_wells # rubocop:todo Metrics/AbcSize
-    if wells_to_fail.empty?
-      redirect_to(
-        limber_plate_path(params[:id]),
-        notice: 'No wells were selected to fail' # rubocop:todo Rails/I18nLocaleTexts
-      )
-    else
+    return redirect_to(limber_plate_path(params[:id]), notice: t('notices.no_wells_selected')) if selected_wells.empty?
+
+    begin
       Sequencescape::Api::V2::StateChange.create!(
-        contents: wells_to_fail,
+        contents: selected_wells,
         customer_accepts_responsibility: params[:customer_accepts_responsibility],
         reason: 'Individual Well Failure',
         target_state: 'failed',
         target_uuid: params[:id],
         user_uuid: current_user_uuid
       )
-      redirect_to(
-        limber_plate_path(params[:id]),
-        notice: 'Selected wells have been failed' # rubocop:todo Rails/I18nLocaleTexts
-      )
+      redirect_to(limber_plate_path(params[:id]), notice: t('notices.wells_failed'))
+    rescue StandardError => e
+      log_plate_error(e)
+      redirect_to limber_plate_path(params[:id]), alert: t('errors.messages.fail_wells_failed')
+      nil
     end
   end
 
-  def wells_to_fail
+  def selected_wells
     params.fetch(:plate, {}).fetch(:wells, {}).select { |_, v| v == '1' }.keys
   end
 
+  def process_mark_under_represented_wells # rubocop:todo Metrics/AbcSize
+    return redirect_to(limber_plate_path(params[:id]), notice: t('notices.no_wells_selected')) if selected_wells.empty?
+
+    begin
+      plate = fetch_plate_with_requests(params[:id])
+      mark_selected_wells_under_represented(plate)
+      redirect_to(limber_plate_path(params[:id]), notice: t('notices.wells_marked_under_represented'))
+    rescue StandardError => e
+      log_plate_error(e)
+      redirect_to limber_plate_path(params[:id]), alert: t('errors.messages.mark_wells_under_represented_failed')
+      nil
+    end
+  end
```

Comment:
I think we should refrain from todo-ing Rubocop for new code we add. It does seem that [`Metrics/AbcSize`](https://docs.rubocop.org/rubocop/cops_metrics.html) is too restrictive, but I suggest making it agree with Rubocop standards rather than todo-ing it. 


```suggestion
def process_mark_under_represented_wells
  return redirect_to_no_wells_selected if selected_wells.empty?
  mark_wells_under_represented_and_redirect
end

private

def redirect_to_no_wells_selected
  redirect_to(limber_plate_path(params[:id]), notice: t('notices.no_wells_selected'))
end

def mark_wells_under_represented_and_redirect
  plate = fetch_plate_with_requests(params[:id])
  mark_selected_wells_under_represented(plate)
  redirect_to(limber_plate_path(params[:id]), notice: t('notices.wells_marked_under_represented'))
rescue StandardError => e
  log_plate_error(e)
  redirect_to(limber_plate_path(params[:id]), alert: t('errors.messages.mark_wells_under_represented_failed'))
end
```

(Personally, I don't find your function harder to read or unmaintainable. It's just one of those Rubocop things that makes it too restrictive.)

Reply:
refactored.